### PR TITLE
ref(team-selector): Memoize props

### PR DESCRIPTION
**Pull request by ArthurKnaus**
* Memoize non-atomic props that `TeamSelector` passes to to its child
* `InviteMembersModal`: Expose organization on the context for children that use `withOrganization()`

relates to https://github.com/getsentry/sentry/pull/54931
see comment https://github.com/getsentry/sentry/pull/54931#discussion_r1297225972

### CODE REVIEW

File: static/app/components/modals/inviteMembersModal/index.tsx [hunk 1]
```diff
@@ -52,6 +53,10 @@ class InviteMembersModal extends DeprecatedAsyncComponent<
   InviteMembersModalProps,
   State
 > {
+  static childContextTypes = {
+    organization: SentryTypes.Organization,
+  };
```
Comment (scttcper) [inline static/app/components/modals/inviteMembersModal/index.tsx:58]: we typically pass the organization to the modal when its opened, is that an option?

Comment (scttcper) [reply static/app/components/modals/inviteMembersModal/index.tsx:58]: can we use `OrganizationContext.Provider` instead

Comment (ArthurKnaus) [reply static/app/components/modals/inviteMembersModal/index.tsx:58]: Thx! Good to know that we have that. I would love to use it, though the `TeamSelector` (which is widely used in the app) still uses `withOrganization` which only picks up the org from the legacy context. 🤔 
Options, I see:
1. Adjust `withOrganization` to also use `OrganizationContext `
2. Update `TeamSelector` to use the `OrganizationContext`
3. Remove it from this PR

I would remove it as 1. and 2. have a high risk of introducing bugs and / or need bigger code changes.
It should be its own PR.
Is that fine for you?

Comment (scttcper) [reply static/app/components/modals/inviteMembersModal/index.tsx:58]: nah just keep using what you have

**Review by scttcper — APPROVED** [review]

**Pull request merged**